build: reuse fake core in type checking tests

crisbeto opened the pull request:
Currently we have two fake copies of `@angular/core` in the compiler tests which can be out of sync and cause inconsistent tests. These changes reuse a single copy instead.

devversion reviewed (approved):
Nice nice! only one left 😄

crisbeto committed:
build: reuse fake core in type checking tests

Currently we have two fake copies of `@angular/core` in the compiler tests which can be out of sync and cause inconsistent tests. These changes reuse a single copy instead.

jessicajaniuk commented:
This PR was merged into the repository by commit 153fc61d451d7552a26fd75c9f66275a21a42e07.
